eng/tools/TypeSpecValidation/.gitignore

```diff
@@ -0,0 +1 @@
+package-lock.json
```

Comment:
Is this necessary here? We generally try to avoid having nested .gitignore files. If we need to ignore then we should probably add to the root ignore file or maybe use an option that would not output the lock file.

Reply:
there's still a lock file at the root level, I'm guessing we would want that to update accordingly?

Reply:
Actually thinking about this a little more I suggest we don't ignore the lock file at all and include it in the repo.

Reply:
added lock file